tests/unit/store/createStore.ts

```diff
@@ -370,5 +380,80 @@ registerSuite({
 				assert.deepEqual(result, ids, 'Should have returned all deleted ids');
 			}).then(dfd.resolve);
 		}
+	},
+
+	'async storage': {
+		'async operation should not be done immediately.'(this: any) {
+			const{ dfd, store } = getStoreWithAsyncStorage(this, { put: 21 });
+
+			const start = Date.now();
+			store.add(createData()).then(dfd.callback(function(result: ItemType[]) {
+				const finish = Date.now();
+				assert.isAbove(finish - start, 19 );
+				dfd.resolve();
+			}));
+		},
+		'should complete initial add before subsequent operations'(this: any) {
+			const dfd = this.async(1000);
```

Comment:
why do you need to use `dfd` here with `this.async`? Can't the tests just return a promise?